hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/targettable/TargetConfirmationOptionsComponent.java

```diff
@@ -0,0 +1,184 @@
+        }
+
+        final boolean isAutoConfirmationEnabled = targetConfirmationOptions.isAutoConfirmationEnabled();
+
+        if (isAutoConfirmationEnabled) {
+            final KeyValueDetailsComponent detailsLayout = buildAutoConfirmationDetailsLayout(
+                    targetConfirmationOptions);
+            targetConfirmationOptionsLayout.addComponent(detailsLayout);
+            targetConfirmationOptionsLayout.setExpandRatio(detailsLayout, 1.0F);
+        } else {
+            final Label confirmationRequiredLabel = buildConfirmationRequiredLabel();
+            targetConfirmationOptionsLayout.addComponent(confirmationRequiredLabel);
+            targetConfirmationOptionsLayout.setExpandRatio(confirmationRequiredLabel, 1.0F);
+        }
+
+        final Button button = buildAutoConfirmationToggleButton(targetConfirmationOptions);
+        targetConfirmationOptionsLayout.addComponent(button);
+    }
+
+    private KeyValueDetailsComponent buildAutoConfirmationDetailsLayout(final ProxyTargetConfirmationOptions options) {
+        final List<ProxyKeyValueDetails> values = Arrays.asList(
+                new ProxyKeyValueDetails(AUTO_CONFIRMATION_DETAILS_STATE,
+                        i18n.getMessage("label.target.auto.confirmation.state"),
+                        i18n.getMessage("label.target.auto.confirmation.active")),
+                new ProxyKeyValueDetails(AUTO_CONFIRMATION_DETAILS_INITIATOR,
+                        i18n.getMessage("label.target.auto.confirmation.initiator"), options.getInitiator()),
+                new ProxyKeyValueDetails(AUTO_CONFIRMATION_DETAILS_REMARK,
+                        i18n.getMessage("label.target.auto.confirmation.remark"), options.getRemark()),
+                new ProxyKeyValueDetails(AUTO_CONFIRMATION_DETAILS_ACTIVATEDAT,
+                        i18n.getMessage("label.target.auto.confirmation.activatedat"),
+                        SPDateTimeUtil.getFormattedDate(options.getActivatedAt())));
+
+        final KeyValueDetailsComponent details = new KeyValueDetailsComponent();
+        details.disableSpacing();
+        details.setValue(values);
+
+        return details;
+    }
+
+    private Label buildConfirmationRequiredLabel() {
+        final Label confirmationLabel = new Label(i18n.getMessage("label.target.auto.confirmation.disabled"));
+        confirmationLabel.setStyleName(ValoTheme.LABEL_SMALL);
+        return confirmationLabel;
+    }
+
+    private Button buildAutoConfirmationToggleButton(final ProxyTargetConfirmationOptions options) {
+        final Button toggleAutoConfirmationButton;
+
+        if (options.isAutoConfirmationEnabled()) {
+            toggleAutoConfirmationButton = SPUIComponentProvider.getButton(
+                    UIComponentIdProvider.AUTO_CONFIRMATION_DETAILS_TOGGLE, "",
+                    i18n.getMessage("button.target.auto.confirmation.disable"), ValoTheme.BUTTON_HUGE, true,
+                    FontAwesome.TOGGLE_ON, SPUIButtonStyleNoBorder.class);
```

Comment:
Any other alternative other using the deprecated ?

Reply:
Using the deprecated one.. could not find similar icon in the Vaadin Icons.